test/unit/MemoryTest.php

```diff
@@ -39,9 +41,99 @@ public function getCommonAdapterNamesProvider(): array
 
     public function testThrowOutOfSpaceException()
     {
-        $this->options->setMemoryLimit(memory_get_usage(true) - 8);
+        $this->options->setMemoryLimit($this->getUsedMemory() - 1);
 
         $this->expectException(OutOfSpaceException::class);
         $this->storage->addItem('test', 'test');
     }
+
+    public function testReclaimMemory()
+    {
+        $outOfSpaceExceptionThrown = false;
+        try {
+            $startMemoryAllocatedToPhp = $this->getAllocatedMemory();
+
+            for ($i = 0; $i < 100000; ++$i) {
+                $this->storage->addItem('item' . $i, sha1((string) mt_rand()));
+            }
+
+            $finishMemoryAllocatedToPhp = $this->getUsedMemory();
+
+            $this->assertGreaterThan($startMemoryAllocatedToPhp, $finishMemoryAllocatedToPhp);
+
+            $this->storage->flush();
+
+            $flushedMemoryAllocatedToPhp = $this->getUsedMemory();
+
+            self::assertLessThan($finishMemoryAllocatedToPhp, $flushedMemoryAllocatedToPhp);
+        } catch (OutOfSpaceException $ignore) {
+            $outOfSpaceExceptionThrown = true;
+        }
+
+        self::assertFalse($outOfSpaceExceptionThrown, 'OutOfSpaceException was thrown');
+    }
+
+    public function testReclaimMemoryPr7()
```

Comment:
Please rename this to `testReclaimMemoryIssue5` and add a link to the issue as method docblock

Reply:
I would say drop PR/issue from the test name entirely